plugin/src/main/java/com/denizenscript/denizen/objects/BiomeTag.java

```diff
@@ -294,9 +288,76 @@ else if (attribute.startsWith("water", 2)) {
         // Biomes with no set foliage color already will have their foliage colors based on temperature and humidity of the biome.
         // -->
         tagProcessor.registerTag(ColorTag.class, "foliage_color", (attribute, object) -> {
-            return new ColorTag(ColorTag.fromRGB(object.biome.getFoliageColor()));
+            return ColorTag.fromRGB(object.biome.getFoliageColor());
         });
 
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
+
+            // <--[tag]
+            // @attribute <BiomeTag.temperature_at[<location>]>
+            // @returns ElementTag(Decimal)
+            // @description
+            // Returns the temperature of a specific location in this biome.
+            // If this is less than 0.15, snow will form on the ground when weather occurs in the world and a layer of ice will form over water.
+            // @example
+            // # Gives the player water if they are standing in a warm location.
+            // - if <player.location.biome.temperature_at[<player.location]> > 0.5:
+            //   - give water_bucket
+            // -->
+            tagProcessor.registerTag(ElementTag.class, LocationTag.class, "temperature_at", (attribute, object, param) -> {
+                return new ElementTag(object.biome.getTemperatureAt(param));
+            });
+
+            // <--[tag]
+            // @attribute <BiomeTag.downfall_at[<location>]>

```

Comment:
These should probably have `LocationTag` tags (that automatically use the biome at the location)

Reply:
Added `LocationTag.temperature/downfall_type` - the biome tags are technically a bit redundant now (as you'll generally always check a location's temperature against the biome it's in), but I guess there's no harm in keeping them for any potential special cases